mozzius/expo-dynamic-app-icon
# Support alternative dark and tinted icon variants

## Motivation

https://github.com/user-attachments/assets/7b7232b1-861e-4abc-93a9-8ddb8688d30a

With the latest iOS 18 (and Expo 52), we can officially support dark and tinted variants of our main app icon. This PR brings parity to alternative app icons defined via this library.

This has to use a different approach: instead of linking assets via `CFBundleIconFiles` in the `Info.plist`, we're instead creating `.appiconset`s for each icon type and adding them to the `Images.xcassets` asset catalog. 

> This is the recommended process when creating alternative icons from Xcode: https://developer.apple.com/documentation/xcode/configuring-your-app-icon

## What This Does

- Updates configuration options to support defining `light`, `dark`, and `tinted` variants for iOS:
```tsx
normal: {
  ios: './assets/normal.png'
},
fancy: {
  ios: {
    light: './assets/fancy-light.png',
    dark: './assets/fancy-dark.png',
    tinted: './assets/fancy-tinted.png'
  }
}
```
- Updates Expo plugin to generate universal (1024x) `.appiconset`s for each icon and include them in the `.pbxproj`'s `XCBuildConfiguration` settings
- Updates iOS Expo module to prefix `AppIcon-` when setting and remove the prefix when getting the current icon

## How to Test

1. Clean up the `example` directory of built assets (`rm -rf .expo`, `rm -rf ios`)
2. Build the plugin (`yarn build:plugin`) and the module (`yarn build`)
3. Rebuild the iOS app (`yarn ios`)
    - You might need to restart your simulator too. Sometimes springboard gets mixed up with alternate app icons
4. Click around on the buttons. Confirm you get the app icon change confirmations
5. Change to the `dark` mode configuration. Exit to the home screen and confirm the customization options change the icon variant
